### PR TITLE
use IFelt

### PR DESCRIPTION
**Pull request by MengZhuNAV**
port IFelt to the project and use it in the OpprettFagSak page

### CODE REVIEW

File: src/frontend/komponenter/Fagsak/Opprett/OpprettFagsak.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    return (
+                        <div key={index} className={'opprettbehandling__barn'}>
+                            <Input
+                                label={'Fødselsnummer'}
+                                value={barnsFødselsnummerFelt.verdi}
+                                onChange={event => {
+                                    dispatch({
+                                        payload: {
+                                            fødselsnummer: event.target.value,
+                                            index,
```
Comment (henninghaakonsen) [inline src/frontend/komponenter/Fagsak/Opprett/OpprettFagsak.tsx]: Nice solution to send in the index 👍

File: src/frontend/komponenter/Fagsak/Opprett/typer.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { IFelt } from '../../../typer/felt';
+
+export type IFøldselsnummerFelt = IFelt<string>;
```
Comment (henninghaakonsen) [inline src/frontend/komponenter/Fagsak/Opprett/typer.ts]: I dont think we need to move these to a separate file. They can just be defined in the interface IStore.

File: src/frontend/komponenter/Fagsak/Opprett/validator.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+): IFøldselsnummerListeFelt => {
+    if (fødselsnummerListeFelt.verdi.length === 0) {
+        return feil(fødselsnummerListeFelt, 'Fødselsnummer.TomListe');
+    }
+
+    let firstError = -1;
+    const validatedListeFelt = {
+        ...fødselsnummerListeFelt,
+        verdi: fødselsnummerListeFelt.verdi.map((felt, index) => {
+            const validatedFelt = felt.valideringsFunksjon(felt);
```
Comment (henninghaakonsen) [inline src/frontend/komponenter/Fagsak/Opprett/validator.ts]: I think it should be possible to validate each child's fnr seperatly. Can you try to create an array with IFelt for each child? This way we can render an error message for each fnr.

File: src/frontend/typer/felt.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+export const ok = <T>(felt: IFelt<T>): IFelt<T> => {
```
Comment (henninghaakonsen) [inline src/frontend/typer/felt.ts]: Great solution 🥇

File: src/frontend/komponenter/Fagsak/Opprett/validator.ts (hunk 3 above)
Comment (MengZhuNAV) [reply src/frontend/komponenter/Fagsak/Opprett/validator.ts]: Yes, actually the fødselsnummerListeFelt is defined as IFelt<IFelt<string>[]>, which can hold both a global error message and individual messages for each child's value

File: src/frontend/komponenter/Fagsak/Opprett/typer.ts (hunk 2 above)
Comment (MengZhuNAV) [reply src/frontend/komponenter/Fagsak/Opprett/typer.ts]: Agree, I will do it🙂

File: src/frontend/komponenter/Fagsak/Opprett/OpprettFagsak.tsx (hunk 1 above)
Comment (MengZhuNAV) [reply src/frontend/komponenter/Fagsak/Opprett/OpprettFagsak.tsx]: Takk😁

File: src/frontend/komponenter/Fagsak/Opprett/validator.ts (hunk 3 above)
Comment (henninghaakonsen) [reply src/frontend/komponenter/Fagsak/Opprett/validator.ts]: Oh, my bad 👍

File: src/frontend/typer/felt.ts (hunk 4 above)
Comment (MengZhuNAV) [reply src/frontend/typer/felt.ts]: 😁

**Pull request merged**